File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool {
    var stk = Array<Character>()
    var s: Array<Character> = Array(s)
    for ch in s {
        if ch == "(" {
            stk.append(ch)
        } else if ch == ")" {
            if stk.isEmpty { return false }
            if stk.last! == "(" { stk.removeLast() }
        }
    }
    return stk.isEmpty ? true : false
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<bits/stdc++.h>

using namespace std;

bool solution(string s) {
    int n = 0;
    for(int i = 0; i <s.size(); i++) {
        if (s[i] == '(') n++;
        else if (--n < 0) return false;
    }
    return n == 0;
}